js/schedule-script.js

```
@@ -131,6 +105,47 @@ const updateDateAndTime = () => {
+
+      await getData(scheduleUrl, "stu-schedule", onSuccess, onFail);
+      const jsonData = localStorage.getItem("stu-schedule");
+      const parsedData = JSON.parse(jsonData);
+      resolve(parsedData);
+    } catch (error) {
+      console.error("Error parsing student data:", error);
+      showAlert("Session error", error.message);
+      reject(error);
+    }
+  }
+});
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)